lib/util/Components.js

```diff
@@ -258,6 +258,8 @@ function componentRule(rule, context) {
       var returnsReactCreateElement =
         node[property] &&
         node[property].callee &&
+        node[property].callee.object &&
+        node[property].callee.object.name === 'React' &&
```

Comment:
You're right. I found a way to check for this by using the ModuleScope of the Variable when importing `createElement` like that. However, it involves using Map, and it doesn't seem like I can get away with that with the project setup.